sql/pgwire: add telemetry for pgwire extended protcol

mneverov opened the pull request:
sql/pgwire: add telemetry for pgwire extended protcol

Fixes #65622

This PR adds telemetry counters for parse, bind, describe, and execute commands.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65679)
<!-- Reviewable:end -->

mneverov reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65679)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mneverov)

*[a discussion](https://reviewable.io/reviews/cockroachdb/cockroach/65679#-MaZrlx52sYWjCktrPPg:-MaZrlx52sYWjCktrPPh:b-yure7e) (no related file):*
Note to the reviewer: please let me know if counters for close and flush messages should be added.

---


<!-- Sent from Reviewable.io -->

rafiss reviewed (changes requested):
Good catch. Yes, let's add close/flush telemetry

rafiss commented:
cc @vy-ton

mneverov committed:
sql/pgwire: add telemetry for pgwire extended protcol

Fixes #65622
This PR adds telemetry counters for parse, bind, describe, and execute commands.

Release note: None

rafiss reviewed (approved):
thank you!!

bors r+
